policy-test/tests/e2e_server_authorization.rs

```diff
@@ -0,0 +1,356 @@
+use linkerd_policy_controller_k8s_api::{
+    self as k8s, policy::server_authorization::Client as ClientAuthz, ResourceExt,
+};
+use linkerd_policy_test::{create, create_ready_pod, curl, nginx, with_temp_ns, LinkerdInject};
+
+#[tokio::test(flavor = "current_thread")]
+async fn meshtls() {
+    with_temp_ns(|client, ns| async move {
+        let srv = create(&client, nginx::server(&ns)).await;
+
+        create(
+            &client,
+            server_authz(
+                &ns,
+                "nginx",
+                &srv,
+                ClientAuthz {
+                    mesh_tls: Some(k8s::policy::server_authorization::MeshTls {
+                        identities: Some(vec!["*".to_string()]),
+                        ..Default::default()
+                    }),
+                    ..Default::default()
+                },
+            ),
+        )
+        .await;
+
+        // Create the nginx pod and wait for it to be ready.
+        tokio::join!(
+            create(&client, nginx::service(&ns)),
+            create_ready_pod(&client, nginx::pod(&ns))
+        );
+
+        let curl = curl::Runner::init(&client, &ns).await;
+        let (injected, uninjected) = tokio::join!(
+            curl.run("curl-injected", "http://nginx", LinkerdInject::Enabled),
+            curl.run("curl-uninjected", "http://nginx", LinkerdInject::Disabled),
+        );
+        let (injected_status, uninjected_status) =
+            tokio::join!(injected.exit_code(), uninjected.exit_code());
+        assert_eq!(
+            injected_status, 0,
+            "uninjected curl must fail to contact nginx"
```

Comment:
Is this statement reversed with the one below it? What I mean by that is that I'd expect the message here to be:

```suggestion
            "injected curl must contact nginx"
```

Since the status is 0 and the pod injected? Similarly, below we'd expect the message to related to the uninjected workload?